src/resizing.jl

```diff
@@ -1,236 +1,5 @@
-                            tmp = convert(T, A[$(Ipre...), iA, $(Ipost...)])
-                            out[$(Ipre...), iout, $(Ipost...)]   += oneeighth*tmp
-                            out[$(Ipre...), iout+1, $(Ipost...)] += threeeighths*tmp
-                        end
-                    end
-                    peakfirst = !peakfirst
-                    iout += peakfirst
-                end
-            end
-        end
-        out
-    end
-end
-
-# If we're restricting along dimension 1, there are some additional efficiencies possible
-@generated function _restrict!(out, indout, A, ::NTuple{0,AbstractUnitRange},
-                               indA, indspost::NTuple{Npost,AbstractUnitRange}) where Npost
-    Ipost = [Symbol(:ipost_, d) for d = 1:Npost]
-    quote
-        $(Expr(:meta, :noinline))
-        T = eltype(out)
-        if isodd(length(indA))
-            half = convert(eltype(T), 0.5)
-            quarter = convert(eltype(T), 0.25)
-            @inbounds @nloops $Npost ipost d->indspost[d] begin
-                iout, iA = first(indout), first(indA)
-                nxt = convert(T, A[iA+1, $(Ipost...)])
-                out[iout, $(Ipost...)] = half*convert(T, A[iA, $(Ipost...)]) + quarter*nxt
-                for iA in first(indA)+2:2:last(indA)-2
-                    prv = nxt
-                    nxt = convert(T, A[iA+1, $(Ipost...)])
-                    out[iout+=1, $(Ipost...)] = quarter*(prv+nxt) + half*convert(T, A[iA, $(Ipost...)])
-                end
-                out[iout+1, $(Ipost...)] = quarter*nxt + half*convert(T, A[last(indA), $(Ipost...)])
-            end
-        else
-            threeeighths = convert(eltype(T), 0.375)
-            oneeighth = convert(eltype(T), 0.125)
-            z = zero(T)
-            @inbounds @nloops $Npost ipost d->indspost[d] begin
-                c = d = z
-                iA = first(indA)
-                for iout = first(indout):last(indout)-1
-                    a, b = c, d
-                    c, d = convert(T, A[iA, $(Ipost...)]), convert(T, A[iA+1, $(Ipost...)])
-                    iA += 2
-                    out[iout, $(Ipost...)] = oneeighth*(a+d) + threeeighths*(b+c)
-                end
-                out[last(indout), $(Ipost...)] = oneeighth*c + threeeighths*d
-            end
-        end
-        out
-    end
-end
-
-restrict_size(len::Integer) = isodd(len) ? (len+1)>>1 : (len>>1)+1
-function restrict_indices(r::AbstractUnitRange)
-    f, l = first(r), last(r)
-    isodd(f) && return (f+1)>>1:restrict_size(l)
-    f>>1 : (isodd(l) ? (l+1)>>1 : l>>1)
-end
-restrict_indices(r::Base.OneTo) = Base.OneTo(restrict_size(length(r)))
-function restrict_indices(r::UnitRange)
-    f, l = first(r), last(r)
-    isodd(f) && return (f+1)>>1:restrict_size(l)
-    f>>1 : (isodd(l) ? (l+1)>>1 : l>>1)
-end
+# restrict
+restrict(img::Union{WarpedView, InvWarpedView}) = restrict(OffsetArray(img)) # preserve axes information
```

Comment:
Actually, this should be

```suggestion
restrict(img::Union{WarpedView, InvWarpedView}, dims) = restrict(OffsetArray(img), dims)
```

Test cases for them are needed...